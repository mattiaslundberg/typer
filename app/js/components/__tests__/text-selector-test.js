import React from 'react'
import TextSelector from '../text-selector'
import sd from 'skin-deep'

describe("TextSelectorTest", () => {
  const render = (cb=jest.genMockFn()) => {
    return sd.shallowRender(
      <TextSelector onSelect={cb} options={[{"fulltext": "some full text", "name": "One"}]}/>
    )
  }

  it('Render selectable items', () => {
    let tree = render()
    let instance = tree.getMountedInstance()

    expect(tree.getRenderOutput()).toMatchSnapshot()
  })

  it('Triggers callback event', () => {
    let cb = jest.fn()
    let tree = render(cb)
    let instance = tree.getMountedInstance()

    let evt = {
      target: {
        value: "test",
      },
    }

    instance.onChange(evt)
    expect(cb).toBeCalled()
    expect(cb.mock.calls[0][0].indexOf("test")).not.toBe(-1)
  })

  it('Correctly removes duplicate spaces', () => {
    let tree = render()
    let instance = tree.getMountedInstance()

    let result = instance.prepareText("abc \t\n\n\t  bc  ")
    expect(result.match(/\s{2,}/)).toBe(null)
  })

  describe('Upload', () => {
    it('Render input form', () => {
      let tree = render()
      let instance = tree.getMountedInstance()
      instance.onAddClick()

      expect(tree.getRenderOutput().props.children).toMatchSnapshot()
    })
  })
})
