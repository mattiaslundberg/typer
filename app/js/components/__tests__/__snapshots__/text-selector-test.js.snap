exports[`TextSelectorTest Render selectable items 1`] = `
<div>
  <p>
    <select
      onChange={[Function bound onChange]}>
      <option
        value="">
        Select text
      </option>
      <option
        value="some full text">
        One
      </option>
    </select>
  </p>
  <p>
    <button
      onClick={[Function bound onAddClick]}>
      Add new
    </button>
  </p>
</div>
`;

exports[`TextSelectorTest Upload Render input form 1`] = `
Array [
  <p>
    <label
        htmlFor="name">
        Name
    </label>
    <input
        name="name"
        type="text" />
</p>,
  <p>
    <label
        htmlFor="text">
        Text
    </label>
    <input
        name="text"
        type="text" />
</p>,
  <p>
    <button
        onClick={[Function bound onNewSubmit]}>
        Save
    </button>
    <button
        onClick={[Function bound onNewCancel]}>
        Cancel
    </button>
</p>
]
`;
